File: uLab/Assets/Scripts/UI/MainWindow.cs

using UnityEngine;
using UnityEngine.UI;

using Locke;
using Locke.ui;


public class MainWindow : IWindow
{
	private static string[] buttonNames = new string[9] { "bag", "role", "shop", "tip", "dialog", "close normals", "close all", "show main", "reset" };
	protected override void OnInit()
	{
		for (int i = 0; i < 9; i++)
		{
			Button button = this.FindWidget<Button>("buttons/button" + (i + 1));
			button.transform.GetComponentInChildren<Text>().text = buttonNames[i];
			UIEventListener.Get(button.gameObject).onClick = OnButtonClick;
		}
	}

	protected override void OnExit(IContext context)
	{
	}

	protected override void OnPause(IContext context)
	{
	}

	protected override void OnResume(IContext context)
	{
	}

	private void OnButtonClick(GameObject go)
	{
		int index = int.Parse(go.name.Substring(go.name.Length - 1));
		
		if (index == 1)
		{
			App.uiManager.OpenWindow(GameUI.bag);
		}
		else if (index == 2)
		{
			App.uiManager.OpenWindow(GameUI.role);
		}
		else if (index == 3)
		{
			App.uiManager.OpenWindow(GameUI.shop);
		}
		else if (index == 4)
		{
			App.uiManager.OpenWindow(GameUI.tip);
		}
		else if (index == 5)
		{
			App.uiManager.OpenWindow(GameUI.dialog);
		}
		else if (index == 6)
		{
			App.uiManager.BackToMainWindow();
		}
		else if (index == 7)
		{
			//UIManager.Instance.CloseAllWindows();
		}
		else if (index == 8)
		{
			//UIManager.Instance.CloseAllWindowsExcept(GameUIList.main);
		}
		else if (index == 9)
		{
			App.uiManager.Cleanup();
		}
	}

}

File: uLab/Assets/Lua/UI/MainWindow.lua

local gameObject
local transform

local MainWindow = class("MainWindow", UIBase)
local this = MainWindow

local buttonNames = { "bag", "role", "shop", "tip", "dialog", "show main", "send chat", "??", "reset" }


function MainWindow:OnInit(obj)
	gameObject = obj
	transform = obj.transform
	--LogInfo("Main OnInit")
	for i = 1, 9 do
		local btn = Util.FindGameObject(gameObject, 'buttons/button'..tostring(i))
		LogInfo(btn.name)
		local label = Util.FindComponent(btn, 'Text', "Text")
		label.text = buttonNames[i]
		UIEventListener.SetOnClick(btn, self.onBtnClick, self)
	end
end

function MainWindow:OnEnter()
	
end

function MainWindow:OnExit()
	
end

function MainWindow:OnPause()
	
end

function MainWindow:OnResume()
	
end

function MainWindow:onBtnClick(go)
	local index = tonumber(string.sub(go.name, string.len(go.name)))
	
	if index == 1 then
		gUIManager:OpenWindow(GameUI.bag)
	elseif index == 2 then
		gUIManager:OpenWindow(GameUI.role)
	elseif index == 3 then
		gUIManager:OpenWindow(GameUI.shop)
	elseif index == 4 then
		gUIManager:OpenWindow(GameUI.tip)
	elseif index == 5 then
		gUIManager:OpenWindow(GameUI.dialog)	
	elseif index == 6 then
		gUIManager:BackToMainWindow()
	elseif index == 7 then
		self.SendChat()
	elseif index == 8 then
		--
	elseif index == 9 then
		App.uiManager:Cleanup()
	end
end

function MainWindow:SendChat()
	local login = {
        account = "locke007",
        password = "2333"
    }
    local bytes = protobuf.encode("Lite.Protocol.cgLogin", login)
	App.networkManager:SendBytes(PBX.MsgID.cgLogin, bytes)
	--local msg = login_pb.cgLogin()
	--msg.account = "locke007"
	--msg.password = "2333"
	--App.networkManager:SendString(PBX.MsgID.cgLogin, msg:SerializeToString())
end

return MainWindow